File: AbstractFactory/python-source/factories/BasementFactory.py
from rooms.BasementHardRoom import BasementHardRoom
from rooms.BasementNormalRoom import BasementNormalRoom
from rooms.HardRoom import HardRoom
from rooms.NormalRoom import NormalRoom

from factories.RoomFactory import RoomFactory


class BasementFactory(RoomFactory):
    def create_normal_room(self) -> NormalRoom:
        return BasementNormalRoom()

    def create_hard_room(self) -> HardRoom:
        return BasementHardRoom()

File: JavaSource/src/main/java/ru/akhcheck/patterns/abstractfactory/factories/BasementFactory.java
package ru.akhcheck.patterns.abstractfactory.factories;

import ru.akhcheck.patterns.abstractfactory.rooms.BasementHardRoom;
import ru.akhcheck.patterns.abstractfactory.rooms.BasementNormalRoom;
import ru.akhcheck.patterns.abstractfactory.rooms.HardRoom;
import ru.akhcheck.patterns.abstractfactory.rooms.NormalRoom;

/** Фабрика комнат подвала. */
public class BasementFactory implements RoomFactory {
    @Override
    public NormalRoom createNormalRoom() {
        return new BasementNormalRoom();
    }

    @Override
    public HardRoom createHardRoom() {
        return new BasementHardRoom();
    }
}
